dotnet/src/Connectors/Connectors.Memory.Weaviate/WeaviateMemoryStore.cs

```diff
@@ -0,0 +1,516 @@
+                $"Unable to create collection: {collectionName}, with class name: {className}", e);
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task<bool> DoesCollectionExistAsync(string collectionName, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNullOrEmpty(collectionName, "Collection name is empty");
+
+        string className = ToWeaviateFriendlyClassName(collectionName);
+        this._log.LogTrace("Does collection exist: {0}, with class name: {1}:", collectionName, className);
+
+        using HttpRequestMessage request = GetClassRequest.Create(className).Build();
+        (HttpResponseMessage response, string responseContent) = await this.ExecuteHttpRequestAsync(request, cancellationToken).ConfigureAwait(false);
+
+        // Needs to return a non-404 AND collection name should match
+        bool exists = response.StatusCode != HttpStatusCode.NotFound;
+
+        if (!exists)
+        {
+            this._log.LogTrace("Collection: {0}, with class name: {1}, does not exist.", collectionName, className);
+        }
+        else
+        {
+            GetClassResponse? existing = JsonSerializer.Deserialize<GetClassResponse>(responseContent, s_jsonSerializerOptions);
+            if (existing != null && existing.Description != ToWeaviateFriendlyClassDescription(collectionName))
+            {
+                // ReSharper disable once CommentTypo
+                // Check that we don't have an accidental conflict.
+                // For example a collectionName of '__this_collection' and 'this_collection' are
+                // both transformed to the class name of SKthiscollection - even though the external
+                // system could consider them as unique collection names.
+                if (existing.Description != ToWeaviateFriendlyClassDescription(collectionName))
+                {
+                    throw new WeaviateMemoryException(WeaviateMemoryException.ErrorCodes.CollectionNameConflict, $"Unable to verify existing collection: {collectionName} with class name: {className}");
+                }
+            }
+        }
+
+        return exists;
+    }
+
+    /// <inheritdoc />
+    public async IAsyncEnumerable<string> GetCollectionsAsync([EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        this._log.LogTrace("Listing collections");
+
+        using HttpRequestMessage request = GetSchemaRequest.Create().Build();
+        (HttpResponseMessage _, string responseContent) = await this.ExecuteHttpRequestAsync(request, cancellationToken).ConfigureAwait(false);
+
+        GetSchemaResponse? getSchemaResponse = JsonSerializer.Deserialize<GetSchemaResponse>(responseContent, s_jsonSerializerOptions);
+
+        foreach (GetClassResponse? @class in getSchemaResponse!.Classes!
+                     .Where(@class => IsPossibleSemanticKernelCollection(@class.Class!)))
```

Comment:
Is it necessary that all my collections should start with `SK` prefix? Let's say I have my collection `contacts` created previously, then I decided to use it with Semantic Kernel, with this logic I won't be able to do that. In the same way, I should be able to create new collection with any name I want, and if I want to mark my collections with specific prefix to identify the source of creation, I can do that by myself, if needed.

Reply:
If you name the collections yourself you are responsible for ensuring that any naming conforms to the Weaviate class naming rules. I chose to take this approach to help the users fall into a pit of success where "it just works".

Reply:
No longer prefixing.